docs/spl-docs/evaluation-functions/comparison-conditional-functions.md

```diff
@@ -0,0 +1,177 @@
+```
+
+
+## validate(\<condition\>, \<value\>, ...)
+This function is opposite of case function. It takes in pairs of condition and value and returns the value corresponding to the first condition that evaluated to `false`. 
+
+#### Usage
+- `<condition>` arguments are Boolean expressions that evaluates to `true` or `false`. 
+- `<value>` argument can either be a string or a numeric literal. 
+- If all conditions are `true`, the default value returned is `NULL`.
+- One can use this function with eval and where commands.
+
+#### Example
+The following command checks the field age and based on the condition returns appropriates values.
+For e.x. if value in the field age is not an integer, it would return `"Age is not an integer"`.
+```
+... | eval is_valid=validate(isint(age), "Age is not an integer", 
+                               age < 18 OR age > 120, "Age is not valid")
+```
+
+
+
+## coalesce(\<values\>)
+This function takes in one or more values and returns the first value that is not `NULL`.
+
+#### Usage
+- `<value>` argument can be a field name, string or numeric literal or an expression.
+- If all the values are `NULL`, it returns `NULL`.
+- One can use this function with eval and where commands.
+
+#### Example
+The following command checks for the user identity from fields username, login_id or email and returns the first value that is not `NULL`. Specifically, it checks in order username, login_id, email. If all the three fields have `NULL` values, it returns `Unknown`.
+```
+... | eval user=coalesce(username, login_id, email, "Unknown")
+```
+
+
+
+
+## searchmatch(\<search_str\>)
+This function returns `true` if the record matches for the search string `<search_str>`. Else returns `false`.
+
+#### Usage
+- `<search_str>` argument is a string containing sets of field value pairs like `"field1=value1 field2=value2"`.
+- searchmatch will search for value in the corresponding field. If all the searches for field value pairs are successful, the functions returns `true`, else returns `false`.
+- Values presented in the search string can have wildcard characters `*` or `?`. 
+- wildcard character `*` is used to match multiple characters.
+- wildcard character `?` is used to match a single character.
+- `<search_str>` could also have a single string literal. In this case, this string literal would be searched in all the available fields and if anyone matches the `<search_str>`, it returns `true`, else `false`.
+- One can use this function with eval and where commands.
+
+#### Example
+The following command searches for occurence of string "error" in the field error_msg and also searches for http error codes in range of 500 in http_status field. If both the searches are successful the function returns `true`, else will return `false`. Thus, when used in combination with `if` command, the field error_type gets the value `"Server error"` if the searchmatch is true else would be `"Unknown"`
```

Comment:
typo: `occurence` should be `occurrence`

You should be able to turn on a spell checker in your code editor, which would be good for markdown files

Reply:
nit: these two sentences don't quite make grammatical sense. You can use an LLM to fix them